### ISSUE TITLE
convoy-nfs and convoy-nfs-storagepool instance stcuck in "initializing" state when they were recreated.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#5259]
Rancher Version:v1.1.0-dev5
Docker Version: 1.11.2
Host os - Ubuntu 15.10

Created "convoy-nfs" from catalog - https://github.com/cjellick/rancher-catalog which uses image-rancher/convoy-agent:v0.9.0.
convoy-nfs and convoy-nfs-storagepool containers come up successsfully.
I was able to create services which used data files from convoy-nfs volume driver successfully.

Few hours later , I see that the convoy-nfs and convoy-nfs-storagepool instance in one of the hosts got recreated and is now stuck in "Initializing" state forever.

```
mysql> select id,name,state, health_state, created,removed from instance where name like "%convoy%";
+----+-------------------------------------+---------+--------------+---------------------+---------------------+
| id | name                                | state   | health_state | created             | removed             |
+----+-------------------------------------+---------+--------------+---------------------+---------------------+
| 39 | convoy-nfs_convoy-nfs-storagepool_1 | purged  | unhealthy    | 2016-06-27 20:26:57 | 2016-06-28 02:47:49 |
| 40 | convoy-nfs_convoy-nfs_1             | running | healthy      | 2016-06-27 20:26:57 | NULL                |
| 41 | convoy-nfs_convoy-nfs_2             | purged  | unhealthy    | 2016-06-27 20:26:57 | 2016-06-28 02:47:50 |
| 47 | convoy-nfs_convoy-nfs-storagepool_1 | running | healthy      | 2016-06-27 21:13:57 | NULL                |
| 48 | convoy-nfs_convoy-nfs_1             | running | healthy      | 2016-06-27 21:13:57 | NULL                |
| 49 | convoy-nfs_convoy-nfs_2             | running | healthy      | 2016-06-27 21:13:58 | NULL                |
| 56 | convoy-nfs_convoy-nfs-storagepool_1 | running | initializing | 2016-06-28 02:47:49 | NULL                |
| 57 | convoy-nfs_convoy-nfs_2             | running | initializing | 2016-06-28 02:47:49 | NULL                |
+----+-------------------------------------+---------+--------------+---------------------+---------------------+
11 rows in set (0.00 sec)

```
logs from convoy-nfs_convoy-nfs-storagepool_1:

```
6/27/2016 7:47:52 PMWaiting for metadata
6/27/2016 7:47:52 PMtime="2016-06-28T02:47:52Z" level=info msg="Listening for health checks on 0.0.0.0:10241/healthcheck"
6/27/2016 7:47:52 PMtime="2016-06-28T02:47:52Z" level=info msg="Socket file: /host/var/run/convoy-convoy-nfs.sock"
6/27/2016 7:47:52 PMtime="2016-06-28T02:47:52Z" level=info msg="Initializing event router" workerCount=10
6/27/2016 7:47:52 PMtime="2016-06-28T02:47:52Z" level=info msg="Connection established"
6/27/2016 7:47:57 PMtime="2016-06-28T02:47:57Z" level=debug msg="storagepool event [699dff77-79a7-47b0-b36f-12ef229c4f8f 477c82bc-b017-4666-a323-64a06a38699b]"
```
logs from convoy-nfs_convoy-nfs_2 container:
```
6/27/2016 7:47:52 PMWaiting for metadata
6/27/2016 7:47:53 PMtime="2016-06-28T02:47:53Z" level=info msg="Execing [/usr/bin/nsenter --mount=/proc/16979/ns/mnt -F -- /var/lib/docker/aufs/mnt/5ff014ebd5ccfdcef78512fe690492022367b1660d938c9b4252cae70b576b97/var/lib/rancher/convoy-agent/share-mnt --stage2 /var/lib/rancher/convoy/convoy-nfs-3159a7e1-2596-484f-9dc7-95b93588492b -- /launch volume-agent-nfs-internal 16979]"
6/27/2016 7:47:53 PMRegistering convoy socket at /var/run/convoy-convoy-nfs.sock
6/27/2016 7:47:53 PMMounting at: /var/lib/rancher/convoy/convoy-nfs-3159a7e1-2596-484f-9dc7-95b93588492b/mnt
6/27/2016 7:47:53 PMMounting nfs. Command: mount -t nfs  104.131.14.104:/data4 /var/lib/rancher/convoy/convoy-nfs-3159a7e1-2596-484f-9dc7-95b93588492b/mnt
6/27/2016 7:47:53 PMtime="2016-06-28T02:47:53Z" level=info msg="Listening for health checks on 0.0.0.0:10241/healthcheck"
6/27/2016 7:47:53 PMtime="2016-06-28T02:47:53Z" level=info msg="Got: create-on-docker-mount true"
6/27/2016 7:47:53 PMtime="2016-06-28T02:47:53Z" level=info msg="Got: root /var/lib/rancher/convoy/convoy-nfs-3159a7e1-2596-484f-9dc7-95b93588492b"
6/27/2016 7:47:53 PMtime="2016-06-28T02:47:53Z" level=info msg="Got: drivers [vfs]"
6/27/2016 7:47:53 PMtime="2016-06-28T02:47:53Z" level=info msg="Got: driver-opts [vfs.path=/var/lib/rancher/convoy/convoy-nfs-3159a7e1-2596-484f-9dc7-95b93588492b/mnt]"
6/27/2016 7:47:53 PMtime="2016-06-28T02:47:53Z" level=info msg="Got: ignore-docker-delete true"
6/27/2016 7:47:53 PMtime="2016-06-28T02:47:53Z" level=info msg="Launching convoy with args: [--socket=/host/var/run/convoy-convoy-nfs.sock daemon --create-on-docker-mount=true --root=/var/lib/rancher/convoy/convoy-nfs-3159a7e1-2596-484f-9dc7-95b93588492b --drivers=vfs --driver-opts=vfs.path=/var/lib/rancher/convoy/convoy-nfs-3159a7e1-2596-484f-9dc7-95b93588492b/mnt --ignore-docker-delete=true]"
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Found existing config. Ignoring command line opts, loading config from /var/lib/rancher/convoy/convoy-nfs-3159a7e1-2596-484f-9dc7-95b93588492b" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg= driver=vfs driver_opts=map[vfs.path:/var/lib/rancher/convoy/convoy-nfs-3159a7e1-2596-484f-9dc7-95b93588492b/mnt] event=init pkg=daemon reason=prepare root="/var/lib/rancher/convoy/convoy-nfs-3159a7e1-2596-484f-9dc7-95b93588492b"
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg= driver=vfs event=init pkg=daemon reason=complete
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering POST, /volumes/umount" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering POST, /snapshots/create" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering POST, /backups/create" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering POST, /volumes/create" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering POST, /volumes/mount" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering DELETE, /snapshots/" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering DELETE, /backups" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering DELETE, /volumes/" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering GET, /info" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering GET, /volumes/list" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering GET, /volumes/" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering GET, /snapshots/" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering GET, /backups/list" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering GET, /backups/inspect" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering plugin handler POST, /VolumeDriver.Path" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering plugin handler POST, /VolumeDriver.Get" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering plugin handler POST, /VolumeDriver.List" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering plugin handler POST, /Plugin.Activate" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering plugin handler POST, /VolumeDriver.Create" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering plugin handler POST, /VolumeDriver.Remove" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering plugin handler POST, /VolumeDriver.Mount" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=debug msg="Registering plugin handler POST, /VolumeDriver.Unmount" pkg=daemon
6/27/2016 7:47:54 PMtime="2016-06-28T02:47:54Z" level=warning msg="Remove previous sockfile at /host/var/run/convoy-convoy-nfs.sock" pkg=daemon
```

**Comment by sangeethah** [comment]
@cjellick: convoy-nfs containers are not pingable using their ip from Network Agent of another host which  may be why containers are stuck in "Initializing" state .

**Comment by cjellick** [comment]
Can close, right @sangeethah ?

**Comment by deniseschannon** [comment]
Per @sangeethah and @cjellick, this was due to cross host networking issues that have been resolved in v1.1.0-rc3.

**Comment by jalona** [comment]
Containers are stuck in "Initializing" state

Centos 7
docker-common-1.10.3-46.el7.centos.10.x86_64
Rancher	v1.1.2
Cattle	v0.165.8
User Interface	v1.1.12
Rancher Compose	v0.8.6

convoy-agent:v0.9.0

convoy-nfs_convoy-nfs_1
```

19.08.2016 10:54:42Waiting for metadata
19.08.2016 10:54:43time="2016-08-19T07:54:43Z" level=info msg="Execing [/usr/bin/nsenter --mount=/proc/30084/ns/mnt -F -- /var/lib/docker/devicemapper/mnt/5ab8f04004806f012b01f1775bb4725806c10ba1299dc263f52bd57a07931ff7/rootfs/var/lib/rancher/convoy-agent/share-mnt --stage2 /var/lib/rancher/convoy/convoy-nfs-3c2361d1-f956-48d4-a65a-e67819b63570 -- /launch volume-agent-nfs-internal 30084]"
19.08.2016 10:54:44Registering convoy socket at /var/run/convoy-convoy-nfs.sock
19.08.2016 10:54:44Mounting at: /var/lib/rancher/convoy/convoy-nfs-3c2361d1-f956-48d4-a65a-e67819b63570/mnt
19.08.2016 10:54:44Mounting nfs. Command: mount -t nfs -o nfsvers=4 it-portal.homecredit.ru:/ /var/lib/rancher/convoy/convoy-nfs-3c2361d1-f956-48d4-a65a-e67819b63570/mnt
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=info msg="Listening for health checks on 0.0.0.0:10241/healthcheck"
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=info msg="Got: root /var/lib/rancher/convoy/convoy-nfs-3c2361d1-f956-48d4-a65a-e67819b63570"
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=info msg="Got: drivers [vfs]"
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=info msg="Got: driver-opts [vfs.path=/var/lib/rancher/convoy/convoy-nfs-3c2361d1-f956-48d4-a65a-e67819b63570/mnt]"
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=info msg="Got: ignore-docker-delete true"
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=info msg="Got: create-on-docker-mount true"
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=info msg="Launching convoy with args: [--socket=/host/var/run/convoy-convoy-nfs.sock daemon --root=/var/lib/rancher/convoy/convoy-nfs-3c2361d1-f956-48d4-a65a-e67819b63570 --drivers=vfs --driver-opts=vfs.path=/var/lib/rancher/convoy/convoy-nfs-3c2361d1-f956-48d4-a65a-e67819b63570/mnt --ignore-docker-delete=true --create-on-docker-mount=true]"
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Creating config at /var/lib/rancher/convoy/convoy-nfs-3c2361d1-f956-48d4-a65a-e67819b63570" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg= driver=vfs driver_opts=map[vfs.path:/var/lib/rancher/convoy/convoy-nfs-3c2361d1-f956-48d4-a65a-e67819b63570/mnt] event=init pkg=daemon reason=prepare root="/var/lib/rancher/convoy/convoy-nfs-3c2361d1-f956-48d4-a65a-e67819b63570"
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg= driver=vfs event=init pkg=daemon reason=complete
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering GET, /info" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering GET, /volumes/list" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering GET, /volumes/" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering GET, /snapshots/" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering GET, /backups/list" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering GET, /backups/inspect" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering POST, /volumes/create" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering POST, /volumes/mount" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering POST, /volumes/umount" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering POST, /snapshots/create" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering POST, /backups/create" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering DELETE, /snapshots/" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering DELETE, /backups" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering DELETE, /volumes/" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering plugin handler POST, /VolumeDriver.Path" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering plugin handler POST, /VolumeDriver.Get" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering plugin handler POST, /VolumeDriver.List" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering plugin handler POST, /Plugin.Activate" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering plugin handler POST, /VolumeDriver.Create" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering plugin handler POST, /VolumeDriver.Remove" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering plugin handler POST, /VolumeDriver.Mount" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=debug msg="Registering plugin handler POST, /VolumeDriver.Unmount" pkg=daemon
19.08.2016 10:54:44time="2016-08-19T07:54:44Z" level=warning msg="Remove previous sockfile at /host/var/run/convoy-convoy-nfs.sock" pkg=daemon
```

convoy-nfs_convoy-nfs-storagepool_1
```
19.08.2016 10:54:50Waiting for metadata
19.08.2016 10:54:50time="2016-08-19T07:54:50Z" level=info msg="Listening for health checks on 0.0.0.0:10241/healthcheck"
19.08.2016 10:54:50time="2016-08-19T07:54:50Z" level=info msg="Socket file: /host/var/run/convoy-convoy-nfs.sock"
19.08.2016 10:54:50time="2016-08-19T07:54:50Z" level=info msg="Initializing event router" workerCount=10
19.08.2016 10:54:50time="2016-08-19T07:54:50Z" level=info msg="Connection established"
19.08.2016 10:54:55time="2016-08-19T07:54:55Z" level=debug msg="storagepool event [dca9bdfc-b2ed-4fc7-98cb-0d5030b0729b 2900d904-5f79-41bd-ada2-b52193a45ee0]"
```
